lib/icu.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+if (!process.binding('config').hasIntl)
+  throw new Error('This Node.js binary was built without ICU support.');
+
+const Buffer = require('buffer').Buffer;
+const normalizeEncoding = require('internal/util').normalizeEncoding;
+const icu = process.binding('icu');
+
+// Maps the supported transcoding conversions. The top key is the from_enc,
+// the child key is the to_enc. The value is the transcoding function to.
+const conversions = {
```

Comment:
sorry. what I meant is does it handle the variations of supported APIs (e.g. should still probably support `'binary'` for the time being; and also both `utf8` and `utf-8`).